domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1596,91 +1595,210 @@ public TotaledResults(ResultSet set, Integer total) {
   }
 
   /**
-   * Return query results as a stream.
-   *
+   * Streamed GET with CQLWrapper (T variant, no facets)
    * @param <T>
    * @param table
    * @param entity
+   * @param fieldName usually "jsonb"
+   * @param filter usually CQL query
+   * @param returnIdField
+   * @param distinctOn may be null
+   * @param streamHandler called for each record
+   * @param replyHandler called when query is complete
+   * @deprecated This function is deprecated because it is impossible to avoid it has
+   * no callback for the case where the HTTP headers are generated.
+   * Furthermore, the hit count is not passed.
+   * Thus, the only option, is to use the replyHandler to produce this.
+   * Use streamGet with {@link PostgresClientStreamResult} instead.
+   * {@link #streamGet(java.lang.String, java.lang.Object, java.lang.String, org.folio.rest.persist.cql.CQLWrapper, boolean, java.lang.String, io.vertx.core.Handler, io.vertx.core.Handler)}
+   */
+  @Deprecated
+  @SuppressWarnings({"unchecked", "squid:S00107"})
+  public <T> void streamGet(String table, T entity, String fieldName,
+    CQLWrapper filter, boolean returnIdField, String distinctOn,
+    Handler<T> streamHandler, Handler<AsyncResult<Void>> replyHandler) {
+
+    Class<T> clazz = (Class<T>) entity.getClass();
+    streamGet(table, clazz, fieldName, filter, returnIdField, distinctOn,
+      res -> {
+        if (res.failed()) {
+          replyHandler.handle(Future.failedFuture(res.cause()));
+          return;
+        }
+        PostgresClientStreamResult<T> streamResult = res.result();
+        streamResult.handler(streamHandler);
+        streamResult.endHandler(x -> replyHandler.handle(Future.succeededFuture()));
```

Comment:
Can you add

            streamResult.exceptionHandler(e -> replyHandler.handle(Future.failedFuture(e)));